### DISCUSSION TITLE
The licence validation error, 'Encrypted email address does not match [. .]' is unhelpful

### DISCUSSION THREAD

**Discussion by LinuxOnTheDesktop** [RealRaven2000/QuickFolders#565] [Announcements]
Just now, on QF 5.8.3 (and on Linux, if that matters), I tries to validate a license. The license text has an email address near its start. Now, I copied the license text to the clipboard but by accident I omitted part of the email address. On trying to validate the address, QF said this (though by mistake my screenshot is missing a few pixels to the left):

![image](https://user-images.githubusercontent.com/47056543/157995617-fe97f47c-c1f2-4f51-863f-1be727266bba.png)

That second message - 'Encrypted email address does not match readable one' is fairly boggling. So perhaps that message should be altered (or simply removed).

**Comment by RealRaven2000** [comment]
:smile: don't worry about it - it's actually true and will only be shown to people who "tamper" with the license key. The key consists of 4 components, product prefix, expiry date, email address and encrypted string. The encrypted string contains the first 3 components but encrypted with an RSA algorithm that QuickFolders can easily decrypt. If the email doesn't match (because the user tried to change his address) then this is the message you see. It helps me specifically when you send a screenshot to support.

**Comment by LinuxOnTheDesktop** [comment]
Hmm. As demonstrated here, accidental mangling of the key can generate the message and, thereby, the baffling of legitimate users.
